src/builder/QuarkBuilder.sol

```diff
@@ -367,6 +424,29 @@ contract QuarkBuilder {
                 if (payment.isToken && Strings.stringEqIgnoreCase(payment.currency, assetSymbol)) {
                     aggregateMaxCosts += PaymentInfo.findMaxCost(payment, chainAccountsList[i].chainId);
                 }
+
+                // If the asset has wrapper counterpart and can locally wrap/unwrap, accumulate the balance of the the counterpart
+                if (TokenWrapper.hasWrapperContract(chainAccountsList[i].chainId, assetSymbol)) {
```

Comment:
Originally wanted this to be a bit smarter as it can help wrap/unwrap token at srcChain if needed. 
Since under this assumption it shouldn't be too hard. As converting all tokens across supported chains to be TransferIntent's token then bridge all into dstChain. 

But if we decided to go with above comments direction (as trim off all briding related, and assume auto wrap/unwrap only exist in one dst chain) then yes we should remove this blocks as well.

Reply:
removed in 3fd23de5ec2e279e89218e1d907ebc2d6b6fed89
If needed in future, should be easy to add it back.